### DISCUSSION TITLE
Unable to import a npm module

### DISCUSSION THREAD

**Discussion by lakshmajee** [remix-run/remix#4866] [Q&A]
Hi all,

I am trying to use an npm library within `example.server.ts` file but while building it is throwing the below error.

```
error TS1323: Dynamic imports are only supported when the '--module' flag is set to 'es2020', 'es2022', 'esnext', 'commonjs', 'amd', 'system', 'umd', 'node16', or 'nodenext'.

279   const { default: plivo } = await import("plivo");
                                       ~~~~~~~~~~~~~~~


Found 1 error.
```

and the contents of `exampl.server.ts` looks like below

```ts
export async function doSomeWork() {
        const { default: plivo } = await import("plivo");

       // This code is running
}
```

I have tried to change the `module` and `target`, both to `esnext`. but **none worked** so far.

Could someone help me !

**Comment by sergiodxa** [comment]
You need to change the `module` value in the `tsconfig.json` of your app.

But even with the TS error dynamic imports should work.

**Reply by lakshmajee** [reply]
Thank you @sergiodxa for quick reply,
Yep, I have mentioned that at the end of the description.
Are there any solutions or suggestions?

**Reply by sergiodxa** [reply]
Could you provide a reproduction? When I tried this I did the change and it worked

**Reply by lakshmajee** [reply]
Created a repo that reproduces the above-mentioned problem, which can be found [here](https://github.com/awkward-minion/dynamic-import-problem)

### Steps to reproduce

1. clone [repository](https://github.com/awkward-minion/dynamic-import-problem)

    ```bash
     git clone https://github.com/awkward-minion/dynamic-import-problem
    ```

2.  install dependencies

    ```bash
     npm i
    ```

3.  Run the below script, to see `typescript` typing error about **dynamic import**
     
    ```bash
     npm run typecheck
    ```
4. we can refer to GitHub actions [here](https://github.com/awkward-minion/dynamic-import-problem/actions/runs/3715277319/jobs/6300242276), for a quick check

> Note: This repository was created from [indie stack](https://github.com/remix-run/indie-stack)
